### PR TITLE
Add new blog post for otelsql

### LINKED ISSUE #4046: New Blog Post: Getting Started with otelsql: OpenTelemetry instrumentation for Go SQL
This is a blog post about how to use [otelsql](https://github.com/XSAM/otelsql) to instrument the `database/sql` package of the Go programming language.

### PR DESCRIPTION
**Pull request by XSAM**
resolve #4046 

cc @svrnm

---

**Preview**: https://deploy-preview-4048--opentelemetry.netlify.app/blog/2024/getting-started-with-otelsql/

### CODE REVIEW

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (XSAM) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:50]: I will update the tag after a new release of `otelsql`

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:25]:
```suggestion
In the following, you are going to use docker compose to run an
example from the otelsql repository. This example uses a MySQL
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:27]:
```suggestion
be pushed to the OpenTelemetry Collector, then it shows the trace data on Jaeger and the
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:364]: TIL, that there is a sqlcommenter website, we probably should figure out what to do here since sqlcommenter is part of OTel for a while now (https://github.com/open-telemetry/opentelemetry-sqlcommenter)

**Review by svrnm — CHANGES_REQUESTED** [review]
overall lgtm, a few suggestions

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (XSAM) [reply content/en/blog/2024/getting-started-with-otelsql/index.md:25]: This paragraph specifies the `otel-collector` example that we are going to use. Is there any reason to remove the word `otel-collector`?

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (XSAM) [reply content/en/blog/2024/getting-started-with-otelsql/index.md:364]: Do you mean we should create a sqlcommenter doc inside the website, and use an internal link instead of an external link to do the reference?

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:6]:
```suggestion
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:20]:
```suggestion
otelsql is a wrapper layer for interfaces from `database/sql`. When
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:24]:
```suggestion
In the following, you are going to use [Docker Compose](https://docs.docker.com/compose/) to run an
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [reply content/en/blog/2024/getting-started-with-otelsql/index.md:25]: then you should say "the `otel-collector` example" to make it clearer that this is the name of that example and not just an example that uses the OpenTelemetry Collector

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:40]:
```suggestion
Let's clone the otelsql repository here and run the example and take a look at the most important lines of code.
```

This is the only(?) place in the document where you use the "I ..." form, let's change this for consistency.

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (theletterf) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:199]:
```suggestion
This `main` function is pretty straightforward. It initializes a connection with
the `otel-collector` service, which is used by the tracer provider and the
meter provider. Then, it configures the tracer provider and meter provider with
the `connection` and a shutdown method, which ensures the telemetry data can be
pushed to the `otel-collector` service correctly before exiting the application.
After finishing setting up the OpenTelemetry client, it invokes the `connectDB`
method to use the otelsql library to interact with the MySQL database. Let's
look at the details here.
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (theletterf) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:229]:
```suggestion
Instead of using the [`sql.Open`](https://pkg.go.dev/database/sql#Open) method
that Go provides, we use
[`otelsql.Open`](https://pkg.go.dev/github.com/XSAM/otelsql#Open) to create an
[`sql.DB`](https://pkg.go.dev/database/sql#DB) instance. The `sql.DB` instance
returned by `otelsql.Open` is a wrapper that transfers and instruments all DB
operations to the underlying `sql.DB` instance (created by `sql.Open`). When
users send SQL queries with this wrapper, `otelsql` can see the queries and
use the OpenTelemetry client to generate telemetry.
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (theletterf) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:238]:
```suggestion
Besides using `otelsql.Open`, `otelsql` provides three additional ways to
initialize instrumentation: `otelsql.OpenDB`, `otelsql.Register`, and
`otelsql.WrapDriver`. These additional methods cover different use cases, as
some database drivers or frameworks don't provide a direct way to create
`sql.DB`. Sometimes, you might need these additional methods to manually
create a `sql.DB` and push it to those database drivers. You can check
[examples on the otelsql document](https://pkg.go.dev/github.com/XSAM/otelsql#pkg-examples)
to learn how to use these methods.
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (theletterf) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:243]:
```suggestion
Moving on, we use `otelsql.RegisterDBStatsMetrics` to register metrics data from
`sql.DBStats`. The metrics recording process runs in the background and
updates the value of the metric when needed after the registration, so we don't
need to worry about creating an individual thread for this.
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (theletterf) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:287]:
```suggestion
After this method, the `client` application finishes and exits. They are
the most important lines of code for understanding the example.
```

Why?

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (theletterf) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:292]:
```suggestion
complicated to see how it will be used in a real-world scenario.
```

**Review by theletterf — CHANGES_REQUESTED** [review]
I love these in-depth posts! Sent some suggestions.

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (chalin) [reply content/en/blog/2024/getting-started-with-otelsql/index.md:2]: Nit: once the colon is gone, you can drop the single quotes too!

**Comment by XSAM** [comment]
@svrnm @theletterf @chalin Thank you for the careful review! I've resolved all the comments. PTAL.

**Review by theletterf — APPROVED** [review]
Thank you!

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@chalin can you take another look if all your change requested have been addressed as needed?

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (svrnm) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:5]:
```suggestion
date: 2024-03-04
```

Please keep this suggestion open such that we set the right publication day before merging

File: .cspell.yml
Comment (chalin) [inline .cspell.yml:174]: These should go in the page-local dictionary IMHO. WDYT @svrnm

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (chalin) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:3]:
```suggestion
  Getting started with otelsql, the OpenTelemetry instrumentation for Go SQL
```

File: content/en/blog/2024/getting-started-with-otelsql/index.md
Comment (chalin) [inline content/en/blog/2024/getting-started-with-otelsql/index.md:10]:
```suggestion
[`database/sql`](https://pkg.go.dev/database/sql) library of the Go programming
```

**Review by chalin — APPROVED** [review]
I won't have the time to do a full review. I've made a few (nitty) drive-by comments. RSLGTM.

File: .cspell.yml
Comment (svrnm) [reply .cspell.yml:174]: sqlcommenter might eventually land there, but yes, they should be page local, I missed that one, thanks!

@XSAM add those terms to the cspell:ignore in the front matter of your blog post, see existing blog posts how that works if you do not have one yet.

**Comment by chalin** [comment]
We got 3 approvals, and the date was updated to today. Thanks all! Merging.

**Pull request merged**
